config/src/config_wrapper.cpp

```diff
@@ -18,26 +18,198 @@
+        }                                                                                          \
+    }
+
+#define LIST_GETTER(p_type, list_type, from, key)                                                  \
+    void key(int32_t num, p_type arr)                                                              \
+    {                                                                                              \
+        if (!loaded) {                                                                             \
+            load();                                                                                \
+        }                                                                                          \
+        try {                                                                                      \
+            auto tmp = from.get<std::vector<list_type>>(#key);                                     \
+            for (int i = 0; i < num; i++) {                                                        \
+                arr[i] = tmp.at(i);                                                                \
+            }                                                                                      \
+        }                                                                                          \
+        catch (const std::exception &e) {                                                          \
+            log << e.what();                                                                       \
+        }                                                                                          \
+    }
+
+bool loaded = false;
 
 config::Config static_config;
+config::Config dynamic_config;
+
+void load()
+{
+    if (loaded) {
+        return;
+    }
+
+    static_config.load_from_file("static_config.json");
+    dynamic_config.load_from_file("dynamic_config.json");
+
+    loaded = true;
+}
+
+CONFIG_GETTER(int32_t, int, static_config, number_of_stations, 0);
+CONFIG_GETTER(int32_t, int, static_config, number_of_end_stations, 0);
+CONFIG_GETTER(int32_t, int, static_config, number_of_robots, 0);
+CONFIG_GETTER(int32_t, int, static_config, number_of_waypoints, 0);
+CONFIG_GETTER(int32_t, int, static_config, waypoint_passing_time, 0);
+CONFIG_GETTER(int32_t, int, static_config, station_passing_time, 0);
+CONFIG_GETTER(double, double, static_config, uncertainty, 0.0);
+
+CONFIG_GETTER(int32_t, int, dynamic_config, current_station, 0);
+CONFIG_GETTER(int32_t, int, dynamic_config, current_waypoint, 0);
+CONFIG_GETTER(int32_t, int, dynamic_config, destination, 0);
+
+void endstation(int32_t number_of_stations, int8_t *arr)
+{
+    load();
+    try {
+        static auto tmp = static_config.get<std::vector<bool>>("endstation");
+        for (int i = 0; i < number_of_stations; i++) {
+            arr[i] = tmp.at(i);
+        }
+    }
+    catch (const std::exception &e) {
```

Comment:
Måske? Ved ikke om der er nogen interesse i at propagere andre exception typer. 
```suggestion
    catch (const std::out_of_range &e) {
```

Reply:
Har samme holdning til de andre catches på arrays her.

Reply:
Vi skal også fange exceptions fra JSON parsing.

Reply:
Kan vi ikke bare have flere catches? Så kan vi få brugbare fejlbeskeder?

Reply:
Det gør du allerede med e.what().

Reply:
Hvad er så pointen med nogensinde at have mere end en catch?